Drop NodeGit

tad-lispy opened the pull request:
Use standard git binary instead.

This introduces an external dependency on `git` command. It's seems to be shortest path to fix long standing issues #5 and #7.

**How to test**

For now manual tests are required. This new code is not yet released to NPM so in order to test in your project, do:

``` sh
git clone https://github.com/lzrski/npm-git-install.git
cd npm-git-install
git checkout drop-node-git
npm link
```

then in a directory of your project with git dependencies:

``` sh
npm link npm-git-install
npm-git install
```

> **Important:** that the CLI command is now `npm-git install` (there is a space between `npm-git` and `install`, where a dash `-` used to be). You may have to update `scripts` section in  your `package.json`.

Change of a CLI command was done in preparation of new subcommand: `shrinkwrap`. See discussion here: lambdatastic/npm-git-install#3

Big thanks to @lambdatastic, @racerpeter and @jrolfs for setting me on the right path.

lambdatastic committed:
Bump nodegit to 0.14

lambdatastic committed:
Merge pull request #1 from lambdatastic/bump-nodegit

Bump nodegit to 0.14

lambdatastic committed:
0.1.4

Commit:
Make it not use NPM Git (circle doesn't like that)

lambdatastic committed:
Merge pull request #2 from racerpeter/patch-1

Make it not use NPM Git (circle doesn't like that)

lambdatastic committed:
Merge master back into local

lambdatastic committed:
0.1.5

lambdatastic committed:
0.1.6

jrolfs committed:
Checkout specified revision after cloning repository

jrolfs committed:
Consolidate `stdio` option into local variable

jrolfs committed:
Use consistent inline object literal spacing

jrolfs committed:
Add whitespace after `temp.track` invocation

jrolfs committed:
0.1.7

Commit:
Add shrinkwrap capability

Introduces two commands for npm-git-install, install, and shrink-wrap

Commit:
Fix a few bugs

Commit:
Update ready

tad-lispy committed:
Readme revisioned

Some clarifications. The order of parameters for functions is reversed to functional (data as last argument) style. Also the CLI examples use npm-git as a name for executable. None of the above is implemented yet, but i'm on it.

tad-lispy committed:
Reverted parameters options

Also rephrased code to more idiomatic coffee-script here and there.

tad-lispy committed:
Change the CLI name

tad-lispy committed:
WIP: Install command refactored, shrinkwrap in progress

Going to temporarly drop shrinkwrap related code and make a release without it to address two currently open issues.

tad-lispy committed:
Temporarily remove shrinkwrap related logic

To release a fix to #5 and #7.

The shrinkwrap will be reintroduced in separate PR.

jrolfs commented on package.json:
```diff
@@ -1,10 +1,10 @@
 {
-  "name": "npm-git-install",
-  "version": "0.1.3",
+  "name": "@celkamada/npm-git-install",
```
Probably want to remove this namespace right 😉?

tad-lispy replied:
Right. Thanks :)

tad-lispy committed:
Remove namespace from package name

lambdatastic commented:
Do we want all these version bumps to be included? It doesn't really affect anything other than ensure parity with my fork, which won't be used by my company once we get all of this sorted upstream.

tad-lispy commented:
You are right, esp. that this release will be a regression compared to your fork, as there is no shrink-wrap feature here yet. I hope to bring it back in the next one, but I would like a separate PR for that. So there really is no parity ATM.

tad-lispy committed:
Un-bump the version number

I'll bump it right before releasing to NPM.

Also we don't need to jump four patch numbers ahead.

danielcrisp commented:
Good work, this has fixed #5 for me 👍

tad-lispy commented:
Happy to hear that @danielcrisp :)

If there are no objections from others I'm going to release it to NPM later today.

racerpeter commented:
lgtm!

tad-lispy commented:
And... it's live 🎉 

https://www.npmjs.com/package/npm-git-install
